### PR TITLE
[error] Let type_check throw TypeError

### PR DESCRIPTION
**Pull request by lin-hitonami**
Related issue = #3422

### CODE REVIEW

File: python/taichi/lang/kernel_impl.py [hunk 1]
```diff
@@ -696,28 +694,7 @@ def wrapped(*args, **kwargs):
...
-                                traceback.print_tb(tb,
-                                                   limit=1,
-                                                   file=sys.stderr)
-                                raise TypeError(str(e)[11:]) from None
-                            # Otherwise, modify the traceback object
-                            tb.tb_next = None
-                            raise TypeError(
-                                str(e)[11:]).with_traceback(tb) from None
-                        tb = tb.tb_next
-                raise
```
Comment (frostming) [inline python/taichi/lang/kernel_impl.py:720]: This part is to tailor the traceback to be more human-readable. You can just change L701-L702 to `except TypeError`

Comment (lin-hitonami) [reply python/taichi/lang/kernel_impl.py:720]: The traceback inside ASTTransformer is now included in the exception now in https://github.com/taichi-dev/taichi/pull/3637, so IIUC it is not necessary anymore?

Comment (frostming) [reply python/taichi/lang/kernel_impl.py:720]: I tried on this branch, the traceback is a bit long to me. but mostly LGTM.

**Review by strongoier — APPROVED** [review]
LGTM!

**Pull request merged**